### PR TITLE
Liqun/a transformer example

### PR DESCRIPTION
**Pull request by liqunfu**
**Description**: Add transformer glue test example

**Motivation and Context**
- to show how to use ORTTrainer to fine-tune a transformer model

### CODE REVIEW

File: orttraining/orttraining/test/python/orttraining_run_glue.py [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+        # configurations not to be changed accoss tests
+        self.max_seq_length = 128
+        self.train_batch_size = 8
+        self.learning_rate = 2e-5
+        self.num_train_epochs = 3.0
+        self.local_rank = -1
+        self.overwrite_output_dir = True
+        self.gradient_accumulation_steps = 1
+        self.data_dir = "/bert_data/hf_data/glue_data/"
+        self.output_dir = os.path.dirname(os.path.realpath(__file__)) + "/glue_test_output/"
```
Comment (thiagocrepaldi) [inline orttraining/orttraining/test/python/orttraining_run_glue.py:67]: ideally paths should be joined through `os.path.join(path1, path2, path3, etc)` to ensure they are portable as opposed to hard-coding "/" on them

File: orttraining/orttraining/test/python/orttraining_transformer_trainer.py [hunk 2]
```diff
@@ -0,0 +1,406 @@
...
+        global_step = 0
+        epochs_trained = 0
+        steps_trained_in_current_epoch = 0
+
+        tr_loss = 0.0
+        logging_loss = 0.0
+        train_iterator = trange(
+            epochs_trained, int(num_train_epochs), desc="Epoch", disable=self.args.local_rank not in [-1, 0],
+        )
+        for epoch in train_iterator:
```
Comment (thiagocrepaldi) [inline orttraining/orttraining/test/python/orttraining_transformer_trainer.py:247]: Shouldnt the epoch be in the outer loop while the data iterator in the inner for loop?

File: orttraining/orttraining/test/python/orttraining_transformer_trainer.py [hunk 3]
```diff
@@ -0,0 +1,406 @@
...
+                            global_step == 1 and self.args.logging_first_step
+                        ):
+                            logs = {}
+                            if self.args.evaluate_during_training:
+                                results = self.evaluate()
+                                for key, value in results.items():
+                                    eval_key = "eval_{}".format(key)
+                                    logs[eval_key] = value
+
+                            loss_scalar = (tr_loss - logging_loss) / self.args.logging_steps
```
Comment (thiagocrepaldi) [inline orttraining/orttraining/test/python/orttraining_transformer_trainer.py:282]: Why using a manual loss scaler instead of the ORTTrainer loss_scaler argument?

Comment (liqunfu) [reply orttraining/orttraining/test/python/orttraining_transformer_trainer.py:282]: this loss_scalar is not the mixed precision loss_scalar. it is a term hugging-face code uses. I prefer to keep the term.

File: orttraining/orttraining/test/python/orttraining_transformer_trainer.py (hunk 2 above)
Comment (liqunfu) [reply orttraining/orttraining/test/python/orttraining_transformer_trainer.py:247]: this:
    for step, inputs in enumerate(epoch_iterator): 
is the data iterator in the inner loop of the epoch outer loop

File: orttraining/orttraining/test/python/orttraining_test_utils.py [hunk 4]
```diff
@@ -38,6 +38,7 @@ def get_lr(args, training_steps, schedule='warmup_poly'):
         return args.learning_rate
 
     schedule_fct = SCHEDULES[schedule]
+    training_steps = training_steps if training_steps < args.max_steps else args.max_steps
```
[the remark this answers is not in the document]
Comment (liqunfu) [reply orttraining/orttraining/test/python/orttraining_test_utils.py:41]: change removed

File: orttraining/orttraining/test/python/orttraining_run_glue.py [hunk 5]
```diff
@@ -0,0 +1,216 @@
+# Code ported from run_glue.py of huggingface transformers
```
[the remark this answers is not in the document]
Comment (liqunfu) [reply orttraining/orttraining/test/python/orttraining_run_glue.py]: updated so it is just similar code
